### DISCUSSION TITLE
Nushell as buildtool

### DISCUSSION THREAD

**Discussion by N4tus** [nushell/nushell#7956] [General]
When stumbling over xtask (and its crates xshell and xflags), I though that all it does was creating commands that mainly invoke other external programs. This is something nu is really good at, so I am tempting to just use nu for my next project instead of xtask, cargo make, or just. 

Are there any areas where nushell might fall short to (except stability)? And what would be features/functionality that would be convenient for such a task that does not exists yet? Or in short is this even a good idea to begin with?

**Comment by tylermumford** [comment]
Thanks for posting this. I was developing an idea for a new command runner, but you mentioned `just`, and it seems like it'll do everything I wanted, but of course, better.

I'll give Nushell and/or just a try as a build tool/command runner. The module system seems to be a good starting point.

```nu
# (project.nu) experiment: Nushell as a command runner
# > use project.nu
# > project build; project test

export def build [] { go build }
export def test  [] { go test }
export def run   [] { go run . }
```

I think Nushell and PowerShell (Core) are really good scripting languages. I never feel like I can trust myself to write a good bash script of more than a few lines; I'm much more comfortable with `nu` and `pwsh`.
